lib/watcher.js

```diff
@@ -267,14 +268,22 @@ class Watcher {
 				// the busy promise to fulfil
 				this.debouncer.cancel();
 				this.clearLogOnNextRun = false;
-				this.rerunAll();
+				if (data === 'u') {
+					this.rerunWithUpdate();
+				} else {
+					this.rerunAll();
+				}
 			});
 		});
 	}
 	rerunAll() {
 		this.dirtyStates = {};
 		this.run();
 	}
+	rerunWithUpdate() {
+		this.dirtyStates = {};
+		this.run(undefined, {updateSnapshots: true});
```

Comment:
Should this update snapshots in *all* tests, or just those tests that just ran? I'm leaning towards the latter. Do you know what Jest settled on?

@lukescott?